### ISSUE TITLE
Agent statsd metrics exporter does not work after dns change

### ISSUE THREAD

**Issue by jakolehm** [kontena/kontena#1653]

**Comment by SpComb** [comment]
Do you mean that the agent's `Statsd` client continues sending the stats to the old IP address, after the configured `kontena grid update --statsd-server=statds.foo` DNS name is updated to resolve to a different IP address?

**Comment by jakolehm** [comment]
Yes, currently user needs to issue `grid update` again so that agent swaps the statsd client to a new one.

**Comment by SpComb** [comment]
This sounds like the typical problem of DNS service discovery not working for most UDP-based applications.

It looks like statsd-ruby also supports a `Statsd.new(protocol: :tcp)`, which might act better in this situation? If the old statsd server goes away, the TCP connection will die, and it will reconnect and re-resolve the configured host: https://github.com/reinh/statsd/blob/v1.3.0/lib/statsd.rb#L428

**Comment by jakolehm** [comment]
Yes, but tcp is not supported in every statsd server (for example telegraf).
